www/include/monitoring/downtime/template/AddDowntime.ihtml

```diff
@@ -154,5 +154,37 @@
 
     initDatepicker("datepicker", "mm/dd/yy", "0");
 
+    function updateEndTime() {
+        var d = new Date($(".datepicker").first().val() + ' ' +  $(".timepicker").first().val());
+        if ($("#duration").val() != '') {
+            var dur = 0;
+            var duration_scale = $("#duration_scale").val();
+            if (duration_scale == 's') {
+                dur = $("#duration").val();
+            } else if (duration_scale == 'm') {
+                dur = $("#duration").val() * 60;
+            } else if (duration_scale == 'h') {
+                dur = $("#duration").val() * 3600;
+            } else if (duration_scale == 'd') {
+                dur = $("#duration").val() * 86400;
+            }
+            dur *= 1000; // Convert to microseconds
+            var e = new Date();
+            e.setTime(d.getTime()+dur);
+            $(".datepicker").last().datepicker("setDate", e);
+            $(".timepicker").last().timepicker("setTime", e.getHours()+':'+e.getMinutes());
+        }
+    }
+
+    $(".datepicker").first().on('change', function(e) {
+        updateEndTime();
+    });
+    $("#duration").on('keypress', function(e) {
+        updateEndTime();
+    });
+    $("#duration_scale").on('change', function(e) {
```

Comment:
only a comment :
are you sure that when a duration is chosen, we need to second datepicker to be set ?
isn't to the BE to convert the duration in a timestamp ?

Reply:
We are updating end date based on start date and duration. When duration changes, end date is updated as well. Of course, user can overwrite end date datepicker (manually updating it), but I think that, when, for example duration is changed from 1 to 2 days, end date should be updated accordingly.